crates/project/src/environment.rs

```diff
@@ -120,19 +129,24 @@ impl ProjectEnvironment {
             let load_direnv = ProjectSettings::get_global(cx).load_direnv.clone();
 
             cx.spawn(|this, mut cx| async move {
-                let mut shell_env = cx
+                let (mut shell_env, error) = cx
                     .background_executor()
                     .spawn({
                         let cwd = worktree_abs_path.clone();
                         async move { load_shell_environment(&cwd, &load_direnv).await }
                     })
                     .await
-                    .ok();
+                    .log_err()
+                    .unzip();
 
                 if let Some(shell_env) = shell_env.as_mut() {
                     this.update(&mut cx, |this, _| {
                         this.cached_shell_environments
-                            .insert(worktree_id, shell_env.clone())
+                            .insert(worktree_id, shell_env.clone());
+
+                        if let Some(error) = error.flatten() {
+                            this.direnv_errors.insert(worktree_id, error);
+                        }
```

Comment:
What if the next call succeeds? Don't you want to remove the entry then?

Reply:
That won't happen as for now there's no way for a worktree to be removed from `cached_shell_environments` so there won't be another call with this `worktree_id`. It's why I think we need #18475 but for now that can't happen